qualtran/surface_code/rotation_cost_model.py

```diff
@@ -0,0 +1,109 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import abc
+import math
+
+from attrs import frozen
+
+from qualtran.surface_code.algorithm_summary import AlgorithmSummary
+
+
+class RotationCostModel(abc.ABC):
+    """Analytical estimate of the complexity of approximating a rotation given an error budget."""
+
+    @abc.abstractmethod
+    def rotation_cost(self, error_budget: float) -> AlgorithmSummary:
+        """Cost of a single rotation."""
+
+    @abc.abstractmethod
+    def prepartion_overhead(self, error_budget) -> AlgorithmSummary:
+        """Cost of preparation circuit."""
+
+
+@frozen
+class LogarithmicModel(RotationCostModel):
```

Comment:
This name makes sense here, where you can clearly see that it's an instance of `RotationCostModel` but I suspect if it shows up outside of the class definition it will lose this critical context. Should probably have rotation in the name somewhere

Reply:
ptal